sway-ir/src/instruction.rs

```diff
@@ -155,6 +155,13 @@ pub enum FuelVmInstruction {
         stored_val: Value,
         key: Value,
     },
+
+    WideBinaryOp {
```

Comment:
Can you explain why we need to have a special set of wide operations at the IR level? If this is a FuelVM specificity, why can't the asm generation figure out using wide ops based on the types being manipulated instead of needing to be told to do so by the IR?

Reply:
> why can't the asm generation figure out using wide ops based on the types being manipulated 

It can´t because WQOP needs three args.

The main difference between `WQOP`, the fuel opcode that we generate, and a normal `ADD` is that `result` and `left` are necessarily ptr; `right` can be both ptr and value.

Reply:
I see, makes sense.